src/main/java/core/basesyntax/OperationStrategyImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        if (!operationHandlers.containsKey(operation)) {
+            throw new IllegalArgumentException("No handler found for operation: " + operation);
+        }
+        return operationHandlers.get(operation);
+    }
+}
```

Comment:
Ensure that the OperationStrategyImpl class is placed in the correct package according to the checklist. It should be in the 'strategy' package to maintain a clean project structure.